widget/tabcontainer.go

```diff
@@ -411,13 +414,78 @@ var _ desktop.Hoverable = (*tabButton)(nil)
 type tabButton struct {
 	BaseWidget
 	hovered      bool
+	focused      bool
+	pressed      bool
 	Icon         fyne.Resource
 	IconPosition buttonIconPosition
 	OnTap        func()
 	Style        ButtonStyle
 	Text         string
 }
 
+// FocusGained is called when the Entry has been given focus.
+func (b *TabContainer) FocusGained() {
+	b.focused = true
+	b.Refresh()
+}
+
+// FocusLost is called when the Entry has had focus removed.
+func (b *TabContainer) FocusLost() {
+	b.focused = false
+	b.Refresh()
+}
+
+// Focused returns whether or not this Entry has focus.
+func (b *TabContainer) Focused() bool {
+	return b.focused
+}
+
+func (b *TabContainer) TypedRune(r rune) {
+
+}
+
+func (b *TabContainer) TypedKey(key *fyne.KeyEvent) {
+	if key.Name == fyne.KeyLeft || key.Name == fyne.KeyUp {
+		if b.current > 0 {
+			b.SelectTabIndex(b.current-1)
+		}
+	} else if key.Name == fyne.KeyRight || key.Name == fyne.KeyDown {
+		if b.current < len(b.Items)-1 {
+			b.SelectTabIndex(b.current+1)
+		}
+	}
+	b.Refresh()
+}
+
+func (b *TabContainer) KeyUp(key *fyne.KeyEvent) {
+	if b.renderer.tabBar==nil {
```

Comment:
Sorry, no - a widget must not manually manipulate the state of it's renderer.
Widgets manage state - renderers draw it.
Surely calling `SelectTab` or similar helper is the right thing to do?

Reply:
This has been fixed now